src/primitives/bitcoin/block.h

```diff
@@ -12,6 +12,22 @@
 #include "serialize.h"
 #include "uint256.h"
 
+/**
+ * If true, block headers contain a payload equal to a Bitcoin Script solution
+ * to a signet challenge as defined in the chain params.
+ */
+extern bool g_solution_blocks;
+/**
+ * If non-zero, defines an enforced size requirement for block header payloads.
+ * It requires that all blocks are of size 80 + (this value) bytes.
+ */
+extern size_t g_solution_block_len;
+ /**
+ * Contains a mapping of hash to signature data for each block header
+ * in signet networks.
+ */
+extern std::map<uint256,std::vector<uint8_t>> g_blockheader_payload_map;
```

Comment:
An in-memory cache that, most importantly, doesn't inflate memory usage when signet feature is off.

This does mean an ever-growing set when on, yes. Wonder what cane be done with this.

Reply:
@kallewoof thoughts?

Reply:
It's ever growing, yep. It could be cached since header sigs are stored along with headers. I haven't thought too much about it, as even a 1 million block chain (that's 2x the lifetime of bitcoin right now) would only have ~100 MB of RAM use.

Reply:
Also when it's a 11-of-15 multisig? Or even larger if this script doesn't have the same script limits as normal tx scripts.

Reply:
Right you would basically store siglen + alpha per block. For a 15 multisig over ECDSA it'd get big quicker, I suspect, but I assume we will eventually switch to Schnorr multisigs long term.